docs/user_guide/choosing_memory_resources.md

```diff
@@ -0,0 +1,289 @@
+
+**Example:**
+```python
+import rmm
+
+# Set async memory resource as default
+rmm.mr.set_current_device_resource(rmm.mr.CudaAsyncMemoryResource())
+
+# Or use rmm.reinitialize for convenience
+rmm.reinitialize(pool_allocator=False, devices=[0])
+```
+
+**When to use:**
+- Default choice for GPU-accelerated applications
+- Multi-stream or multi-threaded applications
+- Applications using multiple GPU libraries (e.g., cuDF + PyTorch)
+- Most production workloads
+
+### CudaMemoryResource
+
+The `CudaMemoryResource` uses `cudaMalloc` directly for each allocation, with no pooling.
+
+**Advantages:**
+- Simple and predictable
+- No fragmentation concerns
+- Memory is immediately returned to the system on deallocation
+
+**Disadvantages:**
+- Slower than pooled allocators due to synchronization overhead
+- Each allocation requires a driver call
+
+**Example:**
+```python
+import rmm
+
+rmm.mr.set_current_device_resource(rmm.mr.CudaMemoryResource())
+```
+
+**When to use:**
+- Simple applications with infrequent allocations
+- Debugging memory issues
+- Testing or benchmarking baseline performance
+
+### PoolMemoryResource
+
+The `PoolMemoryResource` maintains a pool of memory allocated from an upstream resource.
+
+**Advantages:**
+- Fast suballocation from pre-allocated pool
+- Configurable initial and maximum pool sizes
+
+**Disadvantages:**
+- Can suffer from fragmentation (unlike async MR)
+- Pool is not shared across applications
+- Requires careful tuning of pool sizes
+
+**Example:**
+```python
+import rmm
+
+pool = rmm.mr.PoolMemoryResource(
+    rmm.mr.CudaMemoryResource(),  # upstream resource
+    initial_pool_size=2**30,  # 1 GiB
+    maximum_pool_size=2**32   # 4 GiB
+)
+rmm.mr.set_current_device_resource(pool)
+```
+
+**When to use:**
+- Legacy applications (prefer `CudaAsyncMemoryResource` for new code)
+- Specific tuning requirements not met by async MR
+- Wrapping non-CUDA memory sources
+
+**Important**: If using `PoolMemoryResource`, prefer wrapping `CudaAsyncMemoryResource` as the upstream rather than `CudaMemoryResource`:
+
+```python
+# Better: Pool wrapping async MR
+pool = rmm.mr.PoolMemoryResource(
+    rmm.mr.CudaAsyncMemoryResource(),
+    initial_pool_size=2**30
+)
+```
+
+This combines the benefits of both: fast suballocation from RMM's pool and the driver's virtual addressing capabilities.
+
+### ManagedMemoryResource
+
+The `ManagedMemoryResource` uses CUDA unified memory (via `cudaMallocManaged`), allowing memory to be accessible from both CPU and GPU.
+
+**Advantages:**
+- Enables working with datasets larger than GPU memory
+- Automatic page migration between CPU and GPU
+- Simplifies memory management for host/device code
+
+**Disadvantages:**
+- Performance overhead due to page faults and migration
+- Requires careful prefetching for optimal performance
+
+**Example:**
+```python
+import rmm
+
+rmm.mr.set_current_device_resource(rmm.mr.ManagedMemoryResource())
```

Comment:
I'm hesitant to include this because people will copy pasta and see the poor performance that usually causes people to reject managed memory.